src/main/java/org/isf/patient/service/PatientIoOperationRepository.java

```diff
@@ -46,7 +46,7 @@ public interface PatientIoOperationRepository extends JpaRepository<Patient, Int
 	@Query("select p from Patient p where p.code = :id and (p.deleted = :deletedStatus or p.deleted is null)")
 	List<Patient> findAllWhereIdAndDeleted(@Param("id") Integer id, @Param("deletedStatus") char deletedStatus);
 
-	@Modifying
+	@Modifying(clearAutomatically = true)
```

Comment:
Why this was necessary here?

Reply:
Because without that 4 tests fail.

In TestMergePatient.java:

- testMgrMergePatientPatient1MissingInformation()
- testMgrMergePatient()
- testMergePatientHistory()

And in patient/Tests.java:

- testMergePatientHistory()

Reply:
I imagined that, but why they fail? maybe is there an underlying design problem?

Reply:
Maybe; it was low on my list of things to investigate deeply.  I just wanted to get the tests to work :smirk:

Reply:
I'm trying to understand why, despite all those tests are using the same `updateDeleted` method, they are not isolated each other and sharing _apparently_ the same persistence context while, for instance, `testMgrMergePatientMergeNotes` seems working regardless the new annotation. Are the failing methods calling each other in a way I'm not able to see?

Reply:
See https://www.baeldung.com/spring-data-jpa-modifying-annotation and https://stackoverflow.com/questions/43665090/why-do-we-have-to-use-modifying-annotation-for-queries-in-data-jpa for the reasons.   I think we should change it to `@Modifying(clearAutomatically=true, flushAutomatically=true)`

Reply:
it looks like we are facing this "problem" now because of the introduction of `org.springframework.boot.test.context.SpringBootTest` in the new `OHCoreTestCase.java`, which anyway might be a symptom that tests are not correctly written causing uncommited transactions or uncompleted rollbacks (I guess the latter). If the uncommitted transactions and uncompleted rollbacks are exactly what we are testing, then the "problem" is expected and using clearAutomatically and flushAutomatically sounds reasonable, otherwise we should understande the root cause of the fails.

Reply:
IHMO, all the tests need to be reviewed.   Some of them are confusing and convoluted at the very least.   I think we can do that when I rename the test files (discussed previously) and create a new "draft" PR for  a newer and more complete testing structure.